code/modules/surgery/bodyparts/species_parts/genetics_bodyparts.dm

```diff
@@ -0,0 +1,222 @@
+	if(bunny.getStaminaLoss() > 0 || bunny.legcuffed) // cannot reach maximum jump if you have any stamina loss or are legcuffed(bola, bear trap, etc.)
+		bunny_multiplier = min(bunny_multiplier, 5)
+		cast_on.visible_message(span_warning("[cast_on] weakly hops with their genetically-engineered rabbit legs, hampered by their lack of stamina!"))
+		cast_on.balloon_alert_to_viewers("weakly hops")
+	else
+		cast_on.visible_message(span_warning("[cast_on] hops with their genetically-engineered rabbit legs!"))
+		cast_on.balloon_alert_to_viewers("hops")
+	playsound(cast_on, 'sound/effects/arcade_jump.ogg', 75, vary=TRUE)
+
+
+	cast_on.layer = ABOVE_MOB_LAYER
+	if(bunny_multiplier >= 6) // they have committed to the bit, so we will reward it
+		cast_on.pass_flags |= PASSTABLE|PASSGRILLE|PASSWINDOW|PASSMACHINE|PASSSTRUCTURE
+		RegisterSignal(cast_on, COMSIG_MOVABLE_MOVED, PROC_REF(break_glass))
+	else
+		cast_on.pass_flags |= PASSTABLE|PASSGRILLE|PASSMACHINE|PASSSTRUCTURE
+		RegisterSignal(cast_on, COMSIG_MOVABLE_MOVED, PROC_REF(break_grilles))
+	ADD_TRAIT(cast_on, TRAIT_SILENT_FOOTSTEPS, BUNNY_HOP)
+	ADD_TRAIT(cast_on, TRAIT_MOVE_FLYING, BUNNY_HOP)
+
+	cast_on.add_filter(BUNNY_HOP, 2, drop_shadow_filter(color = "#03020781", size = 0.9))
+	var/shadow_filter = cast_on.get_filter(BUNNY_HOP)
+	var/jump_height = 8 + 16 * ((bunny_multiplier - 1) / 5)
+	var/jump_duration = 0.25 SECONDS * bunny_multiplier
+	new /obj/effect/temp_visual/mook_dust(get_turf(cast_on))
+	animate(cast_on, pixel_y = cast_on.pixel_y + jump_height, time = jump_duration / 2, easing = CIRCULAR_EASING|EASE_OUT)
+	animate(pixel_y = initial(owner.pixel_y), time = jump_duration / 2, easing = CIRCULAR_EASING|EASE_IN)
+
+	animate(shadow_filter, y = -jump_height, size = 4, time = jump_duration / 2, easing = CIRCULAR_EASING|EASE_OUT)
+	animate(y = 0, size = 0.9, time = jump_duration / 2, easing = CIRCULAR_EASING|EASE_IN)
+
+	addtimer(CALLBACK(src, PROC_REF(end_jump), cast_on), jump_duration)
+
+/datum/action/cooldown/spell/bunny_hop/proc/break_glass(atom/movable/mover, atom/oldloc, direction)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/bunny = mover
+	for(var/obj/structure/window/window in get_turf(bunny))
+		window.deconstruct(disassembled = FALSE)
+		mover.balloon_alert_to_viewers("smashed through!")
+		bunny.apply_damage(damage = rand(5,15), damagetype = BRUTE, wound_bonus = 15, bare_wound_bonus = 25, sharpness = SHARP_EDGED, attack_direction = get_dir(window, oldloc))
+		new /obj/effect/decal/cleanable/glass(get_step(bunny, bunny.dir))
+	for(var/obj/machinery/door/window/windoor in get_turf(bunny))
+		windoor.deconstruct(disassembled = FALSE)
+		mover.balloon_alert_to_viewers("smashed through!")
+		bunny.apply_damage(damage = rand(5,15), damagetype = BRUTE, wound_bonus = 15, bare_wound_bonus = 25, sharpness = SHARP_EDGED, attack_direction = get_dir(windoor, oldloc))
+		new /obj/effect/decal/cleanable/glass(get_step(bunny, bunny.dir))
+	for(var/obj/structure/grille/grille in get_turf(bunny))
```

Comment:
Any reason not to just call `break_grilles` here? It's not hot enough that it seems like we'd care about proc call overhead.